media/environments/environmentsPanel.js

```diff
@@ -0,0 +1,47 @@
+(function () {
+  const vscode = acquireVsCodeApi();
+
+  const environmentSection = document.getElementById('environment-section');
+
+  window.addEventListener('message', (event) => {
+    const { method, params } = event.data;
+    if (method === 'add-environments') {
+      environmentSection.innerHTML = '';
+      const envList = params.environments;
+
+      envList.forEach((environment) => {
+        const articleElement = document.createElement('article');
```

Comment:
The whole element creation logic can be moved to a separate function. Also use `replaceChildren` if possible. Something like this:
```
environmentSection.replaceChildren(createEnvironmentElements())
```